test/sample_consensus/test_sample_consensus_plane_models.cpp

```diff
@@ -304,6 +304,55 @@ TEST (SampleConsensusModelNormalParallelPlane, RANSAC)
   }
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointT>
+class SampleConsensusModelPlaneTest : private SampleConsensusModelPlane<PointT>
+{
+  public:
+    using SampleConsensusModelPlane<PointT>::SampleConsensusModelPlane;
+    using SampleConsensusModelPlane<PointT>::countWithinDistanceStandard;
+    using SampleConsensusModelPlane<PointT>::countWithinDistanceSSE;
+    using SampleConsensusModelPlane<PointT>::countWithinDistanceAVX;
+};
+
+TEST (SampleConsensusModelPlane, SIMD_countWithinDistance) // Test if all countWithinDistance implementations return the same value
+{
+  const auto seed = static_cast<unsigned> (std::time (nullptr));
+  srand (seed);
+  for (size_t i=0; i<100; i++) // Run as often as you like
+  {
+    // Generate a cloud with 1000 random points
+    PointCloud<PointXYZ> cloud;
+    std::vector<int> indices;
+    cloud.points.resize (1000);
+    for (std::size_t idx = 0; idx < cloud.size (); ++idx)
+    {
+      cloud.points[idx].x = 2.0 * static_cast<float> (rand ()) / RAND_MAX - 1.0;
+      cloud.points[idx].y = 2.0 * static_cast<float> (rand ()) / RAND_MAX - 1.0;
+      cloud.points[idx].z = 2.0 * static_cast<float> (rand ()) / RAND_MAX - 1.0;
+      if (rand () % 2 == 0)
+      {
+        indices.push_back (static_cast<int> (idx));
+      }
+    }
+    SampleConsensusModelPlaneTest<PointXYZ> model (cloud.makeShared (), indices, true);
+
+    // Generate random model parameters
+    Eigen::VectorXf model_coefficients(4);
+    model_coefficients << 2.0 * static_cast<float> (rand ()) / RAND_MAX - 1.0, 2.0 * static_cast<float> (rand ()) / RAND_MAX - 1.0, 2.0 * static_cast<float> (rand ()) / RAND_MAX - 1.0, 0.0;
+    model_coefficients.normalize ();
+    model_coefficients(3) = 2.0 * static_cast<float> (rand ()) / RAND_MAX - 1.0; // Last parameter
+
+    const double threshold = 0.1 * static_cast<double> (rand ()) / RAND_MAX; // threshold in [0; 0.1]
+
+    const auto res_standard = model.countWithinDistanceStandard (model_coefficients, threshold); // Standard
```

Comment:
Maybe put this in `ifdef` guard to ensure that tests compile even when AVX2 isn't available

Reply:
But the tests would compile even if SSE/AVX/AVX2 are not available? Then `countWithinDistanceSSE` and `countWithinDistanceAVX` would be equivalent to `countWithinDistanceStandard`, because the SIMD-part of those functions would be removed and only the call to `countWithinDistanceStandard` would remain.

Reply:
I thought using `countWithinDistanceAVX` will be a compile error if AVX is not available, but looking at the code that's not the case. That would be a better interface since it doesn't make false promises to the end-user. Function is available if functionality is available. What do you think about macros inside vs outside the functions?

Reply:
Please read my comment from January 12 again, where I present 3 possibilities to handle that. I tend to agree to "no false promises".

Reply:
Sorry missed that.

Reply:
Ok, I implemented option 1.